### PR TITLE
b-01220- Integration testing suite using tryorama

### PR DESCRIPTION
**Pull request by JettTech**
- restructures generated output folder structure
- switches to tryorama testing approach for integration testing
 - updates holonix to v0.0.75 and holochain deps to track v0.0.49
- updates UI cosmetics

### CODE REVIEW

File: src/dna-setup/zome-template/Cargo.toml
Comment (zo-el) [inline src/dna-setup/zome-template/Cargo.toml]: Can we create and use the release of `holochain-anchors`

File: src/dna-setup/zome-template/Cargo.toml
Comment (zo-el) [inline src/dna-setup/zome-template/Cargo.toml]: Can we just add `holochain-anchors` to the Cargo.toml and not replace the entire file.

This would need us to update these script every time there is a holochain release.
If we can avoid this we do not need to update these scripts. Maybe only the config.nix once in a while(which is does not need to be updated every time either)

File: src/happ-template/default.nix
Comment (zo-el) [inline src/happ-template/default.nix]: What is in `.holochain/holo/storage/agent-1/test` ?

File: src/happ-template/package.json
Comment (zo-el) [inline src/happ-template/package.json]: We do not need `HC_IGNORE_SIM2H_URL_PROPERTY=true` since we are not pointing it to a sim2h_server in the manifest

File: src/ui-setup/ui-template/package.json
Comment (zo-el) [inline src/ui-setup/ui-template/package.json:46]: Do you need both waait and sleep?
Do you need tape?

**Review by zo-el — CHANGES_REQUESTED** [review]
I have left some small questions

File: src/dna-setup/scripts/generate-dna-zomes.js
Comment (robbiecarlton) [inline src/dna-setup/scripts/generate-dna-zomes.js]: rename `DNA_SETUP_DIR` to `dnaTemplateDir`

File: src/dna-setup/scripts/generate-dna.js
Comment (robbiecarlton) [inline src/dna-setup/scripts/generate-dna.js]: rename to `DNA_TEMPLATE_DIR`

File: src/happ-template/scripts/utils.js
Comment (robbiecarlton) [inline src/happ-template/scripts/utils.js]: remove log

File: src/happ-template/scripts/utils.js
Comment (robbiecarlton) [inline src/happ-template/scripts/utils.js]: remove log

File: src/happ-template/scripts/utils.js
Comment (robbiecarlton) [inline src/happ-template/scripts/utils.js]: remove log

File: src/happ-template/scripts/utils.js
Comment (robbiecarlton) [inline src/happ-template/scripts/utils.js]: remove log

File: src/setup/type-specs/input-schema-description.md
Comment (robbiecarlton) [inline src/setup/type-specs/input-schema-description.md]: lets remove this file

File: src/ui-setup/scripts/generate-ui.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/generate-ui.js]: if `happ` directory is just missing, this should just throw an error

File: src/ui-setup/scripts/generate-ui.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/generate-ui.js]: map over this object, capitalizing keys (and add a comment about why) rather than capitalizing in individual files

File: src/ui-setup/scripts/renderIndex.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderIndex.js]: remove this and above import of ReactModal

File: src/ui-setup/ui-template/src/__integration_tests__/Demo.integration.test.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/__integration_tests__/Demo.integration.test.js]: remove this file

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: remove this layer of templating, just write the string straight to the file.

test should read eg
```
describe('BookPage', () => {
  it('works in connection with the dna', () => {
  })
})
```

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: remove this, not integration

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: rename `s` to `scenario`

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: add a flow of adding a second book, and testing that both books appear and then that when you remove one book, the other is still there.

File: src/ui-setup/scripts/renderTypePage.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePage.js]: This is a work around due to a holochain issue. We should create an issue on github describing that to hc devs

File: src/ui-setup/scripts/renderTypePageTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageTest.js]: just write these descriptions straight to the test

File: src/ui-setup/ui-template/config/jest/cssTransform.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/config/jest/cssTransform.js]: remove this file

File: src/ui-setup/ui-template/config/jest/fileTransform.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/config/jest/fileTransform.js]: remove this file

File: src/ui-setup/ui-template/config/webpackDevServer.config.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/config/webpackDevServer.config.js]: probably dont need this. remove and see if anything breaks. Probably can remove this whole config folder

File: src/ui-setup/ui-template/jest.common.config.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/jest.common.config.js]: remove this file

File: src/ui-setup/ui-template/jest.integration.config.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/jest.integration.config.js]: rmeove

File: src/ui-setup/ui-template/jest.unit.config.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/jest.unit.config.js]: remove

File: src/ui-setup/ui-template/package.json
Comment (robbiecarlton) [reply src/ui-setup/ui-template/package.json:46]: @JettTech says: we can take out `sleep`

File: src/ui-setup/ui-template/src/pages/@type-page.css
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/pages/@type-page.css]: remove `@` from filename

File: src/ui-setup/ui-template/wait-for-conductor.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/wait-for-conductor.js]: remove non startUI branch. call this file and function `waitForConductorAndStartUI`

File: src/ui-setup/ui-template/src/utils/integration-testing/tryoramaIntegration.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/utils/integration-testing/tryoramaIntegration.js]: remove comment

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: try moving this functionality to a `beforeEach`

File: src/ui-setup/scripts/renderTypePageIntegrationTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageIntegrationTest.js]: replace this 

```
if (process.env.REACT_APP_TEST_TYPE === 'integration') {
  orchestrator.run()
} else {
  test.skip('', () => {})
}
```

File: src/ui-setup/ui-template/src/utils/index.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/utils/index.js]: we can remove this

File: src/ui-setup/ui-template/src/utils/integration-testing/tryoramaIntegration.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/utils/integration-testing/tryoramaIntegration.js]: use `process.env.REACT_APP_TEST_INSTANCE` (or whatever thats called)

File: src/dna-setup/zome-template/Cargo.toml
Comment (JettTech) [reply src/dna-setup/zome-template/Cargo.toml]: Already have a PR in holochain-anchors; it's currently awaiting approval from Philip and/or Paul, who have been the most recent maintainers.  I'll follow-up with them.

File: src/happ-template/default.nix
Comment (JettTech) [reply src/happ-template/default.nix]: Residue for the previous integration approach.  I'll remove it. :+1:

**Review by robbiecarlton — CHANGES_REQUESTED** [review]

File: src/ui-setup/ui-template/src/utils/index.js
Comment (robbiecarlton) [inline src/ui-setup/ui-template/src/utils/index.js]: can we rename `alice` to `agent`?

File: src/ui-setup/scripts/renderTypePageTest.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/renderTypePageTest.js]: remove line 42 (`const testDescription`) and this line should read
```
    describe(\`\${name} Entry Page\`, () => 
```

Comment (robbiecarlton) [reply src/ui-setup/scripts/renderTypePageTest.js]: reasoning: the final generated test files should just look like a normal test file.

File: src/ui-setup/scripts/renderIndex.js
Comment (robbiecarlton) [reply src/ui-setup/scripts/renderIndex.js]: We forgot to do this together, but still need to remove all references to ReactModal in the whole repo, this file and anywhere else.

File: src/ui-setup/scripts/generate-ui.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/generate-ui.js]: not sure why this function is named this.

How about: `capitalizeTypeNames`?

File: src/ui-setup/scripts/generate-ui.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/generate-ui.js]: I would call this variable

```capitalizedTypeArray```

`raw` and `map` are both a little misleading in this context

File: src/ui-setup/scripts/generate-ui.js
Comment (robbiecarlton) [inline src/ui-setup/scripts/generate-ui.js]: this (together with lines 26, 27 and 35 (awkward github)) can be simplified to

```
const rawTypeSpec = !typeSpecPath
  ? defaultTypeSpec
  :  JSON.parse(fs.readFileSync(typeSpecPath))
const typeSpec = capitalizeTypeNames(rawTypeSpec)

File: src/setup/type-specs/sample-type-spec.json
Comment (robbiecarlton) [inline src/setup/type-specs/sample-type-spec.json]: Let's put these back how to capitalized. Even though we fix it, I think it makes more sense to have the sample reflect what the typespec should look like

**Review by robbiecarlton — CHANGES_REQUESTED** [review]

File: src/ui-setup/ui-template/src/utils/index.js
Comment (JettTech) [reply src/ui-setup/ui-template/src/utils/index.js]: It's been a DNA testing standard to use alice, but I'm fine with making a separate standard for the UI integration. :+1:

File: src/ui-setup/scripts/renderTypePageTest.js
Comment (JettTech) [reply src/ui-setup/scripts/renderTypePageTest.js]: Did this for the integrated tests, but missed the unit ones. Agreed and updated. :+1:

File: src/ui-setup/scripts/generate-ui.js
Comment (JettTech) [reply src/ui-setup/scripts/generate-ui.js]: I think of an array of key/value pairs as a map, which is why I choose Map for the title - I'm fine with Array, instead though.  I agree that raw is unneeded. :+1:

File: src/ui-setup/scripts/generate-ui.js
Comment (JettTech) [reply src/ui-setup/scripts/generate-ui.js]: Agreed - I was originally keeping this logic in the if/then format to provide a console about referencing the default, but I'll just separate that part from the remaining logic.

**Comment by JettTech** [comment]
@robbiecarlton - Weird... I remember removing the react-modal and running passing tests.. but it somehow ended up back in here.  (Now I'm suspecting that I removed/tested it in the happ itself, but forgot to extend that into the template. Nonetheless.. it's done.

**Review by robbiecarlton — APPROVED** [review]

**Pull request merged**
